src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeLoaderEnvironment.cs

```diff
@@ -119,6 +119,8 @@ internal static void Initialize()
             Instance = new TypeLoaderEnvironment();
             RuntimeAugments.InitializeLookups(new Callbacks());
             s_nativeFormatStrings = new LowLevelDictionary<string, IntPtr>();
+            s_runtimeFieldHandles = new LowLevelDictionary<RuntimeFieldHandleKey, RuntimeFieldHandle>();
```

Comment:
There are many more caches on TypeLoaderEnvironment that need stay alive for the duration of the process - TypeLoaderEnvironment is never unloaded. We only unload the type system context.

The only reason why it's not a static class is because it gives us a chance to actually write unit tests for this, but that ship probably sailed long time ago. At this point we could probably just make it all static, but that should be done with everything. There's great value in consistency because then someone doesn't need to spend 10 minutes figuring out why something is different (only to find out that the answer is "no reason").